starport/cmd/network_chain_init.go

```diff
@@ -0,0 +1,193 @@
+)
+
+const (
+	flagRecover  = "recover"
+	flagMnemonic = "mnemonic"
+	flagKeyName  = "key-name"
+)
+
+// NewNetworkChainInit returns a new command to initialize a chain from a published chain ID
+func NewNetworkChainInit() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "init [launch-id]",
+		Short: "Initialize a chain from a published chain ID",
+		Args:  cobra.ExactArgs(1),
+		RunE:  networkChainInitHandler,
+	}
+
+	c.Flags().Bool(flagRecover, false, "Recover chain account from a mnemonic")
+	c.Flags().String(flagMnemonic, "", "Mnemonic for recovered account")
+	c.Flags().String(flagKeyName, "", "key name for the chain account")
+
+	c.Flags().AddFlagSet(flagNetworkFrom())
+	c.Flags().AddFlagSet(flagSetHome())
+	c.Flags().AddFlagSet(flagSetKeyringBackend())
+	c.Flags().AddFlagSet(flagSetYes())
+
+	return c
+}
+
+func networkChainInitHandler(cmd *cobra.Command, args []string) error {
+	var (
+		recover, _  = cmd.Flags().GetBool(flagRecover)
+		mnemonic, _ = cmd.Flags().GetString(flagMnemonic)
+		keyName, _  = cmd.Flags().GetString(flagKeyName)
+	)
+
+	s := clispinner.New()
+	defer s.Stop()
+
+	var (
+		wg sync.WaitGroup
+		ev = events.NewBus()
+	)
+	wg.Add(1)
+
+	defer wg.Wait()
+	defer ev.Shutdown()
+
+	go printEvents(&wg, ev, s)
+
+	// parse launch ID
+	launchID, err := strconv.ParseUint(args[0], 10, 64)
+	if err != nil {
+		return fmt.Errorf("error parsing launchID: %s", err.Error())
+	}
+
+	// if a chain has already been initialized with this launch ID, we ask for confirmation before erasing the directory
+	chainHome, exist, err := checkChainHomeExist(launchID)
+	if err != nil {
+		return err
+	}
+	if !getYes(cmd) && exist {
+		prompt := promptui.Prompt{
+			Label: fmt.Sprintf("The chain has already been initialized under: %s. Would you like to overwrite the home directory",
+				chainHome,
+			),
+			IsConfirm: true,
+		}
+		s.Stop()
+		if _, err := prompt.Run(); err != nil {
+			fmt.Println("said no")
+			return nil
+		}
+		s.Start()
+	}
+
+	nb, err := newNetwork(cmd, network.CollectEvents(ev))
+	if err != nil {
+		return err
+	}
+
+	// initialize the blockchain from the launch ID
+	initOptions := initOptionWithHomeFlag(cmd, []network.InitOption{network.MustNotInitializedBefore()})
+	sourceOption := network.SourceLaunchID(launchID)
+	blockchain, err := nb.Blockchain(cmd.Context(), sourceOption, initOptions...)
+	if err != nil {
+		return err
+	}
+
+	if err := blockchain.Init(cmd.Context()); err != nil {
+		return err
+	}
+
+	// ask validator information
+	v, err := askValidatorInfo()
+	if err != nil {
+		return err
+	}
+
+	// ask user for mnemonic if not provided and recover option defined
+	if mnemonic == "" && recover {
+		cliquiz.Ask(cliquiz.NewQuestion("Enter your account mnemonic",
+			&mnemonic,
+			cliquiz.DefaultAnswer(""),
+			cliquiz.Required(),
+		))
+	}
+
+	acc, gentxPath, err := blockchain.InitAccount(cmd.Context(), v, keyName, mnemonic)
+	if err != nil {
+		return err
+	}
+	fmt.Printf(`Account created
```

Comment:
Can we use clispinner.Bullet for each item here and also Gentx path?

Reply:
`clispinner.Bullet` doesn't exist, is it something we removed in the past?

Reply:
But we only have the gentx path to show anyway since no account is created at the end

Reply:
it is recently added and now in develop. 👍